components/brave_shields/resources/panel/components/tree-list/index.tsx

```diff
@@ -46,42 +52,98 @@ function groupByOrigin (data: Url[]) {
 
 function TreeList (props: Props) {
   const { siteBlockInfo, setViewType } = React.useContext(DataContext)
-  const mappedData = React.useMemo(() => groupByOrigin(props.data), [props.data])
+  const allowedList = props.allowedList ?? [];
+  const allowedScriptsByOrigin = React.useMemo(() =>
+    groupByOrigin(allowedList), [allowedList])
+
+  const blockedScriptsByOrigin = React.useMemo(() =>
+    groupByOrigin(props.blockedList),
+                  [props.blockedList])
+
+  const handleAllowAllScripts = () => {
+    const origins = props.blockedList.map(entry => new URL(entry.url).origin)
+    getPanelBrowserAPI().dataHandler.allowScriptsOnce(origins)
+  }
+
+  const handleBlockAllScripts = () => {
+    const origins = allowedList.map(entry => new URL(entry.url).origin)
+    getPanelBrowserAPI().dataHandler.blockAllowedScripts(origins)
+  }
+
+  const handleBlockScript = (name: string) => {
+    getPanelBrowserAPI().dataHandler.blockAllowedScripts([new URL(name).origin])
+  }
+
+  const handleAllowScript = allowedList.length === 0 ? undefined
+    : (name: string) => {
+      getPanelBrowserAPI().dataHandler.allowScriptsOnce([new URL(name).origin])
+    }
 
   return (
     <S.Box>
       <S.HeaderBox>
         <S.SiteTitleBox>
           <S.FavIconBox>
-            <img key={siteBlockInfo?.faviconUrl.url} src={siteBlockInfo?.faviconUrl.url} />
+            <img src={siteBlockInfo?.faviconUrl.url} />
           </S.FavIconBox>
           <S.SiteTitle>{siteBlockInfo?.host}</S.SiteTitle>
         </S.SiteTitleBox>
-        <S.Grid>
-          <span>{props.totalBlockedCount}</span>
-          <span>{props.blockedCountTitle}</span>
-        </S.Grid>
       </S.HeaderBox>
-      <S.TreeBox>
-        <div>
-          {[...mappedData.keys()].map((origin, idx) => {
+      <S.Scroller>
+        {allowedList.length > 0 && (
+        <>
+          <ScriptsInfo>
+            <span>{allowedList.length}</span>
+            <span>{props.totalAllowedTitle}</span>
+            <span>{<a href="#" onClick={handleBlockAllScripts}>
+                {getLocale('braveShieldsBlockScriptsAll')}
+              </a>
+            }</span>
+          </ScriptsInfo>
+          <ScriptsList>
+            {[...allowedScriptsByOrigin.keys()].map((origin, idx) => {
+              return (<TreeNode
+                key={origin}
```

Comment:
The key here should be the origin, but in the call below, you are passing the index. Let's ensure consistency by using the same key throughout. If the host is unique, it can serve as the key. 

Additionally, let's ensure consistency in the ordering of props:

1. key
2. origin
3. resourceList
4. onPermissionButtonClick
5. permissionButtonTitle

Reply:
fixed